src/Pair/toPairs.js

```diff
@@ -1,21 +1,21 @@
 /** @license ISC License (c) copyright 2017 original and current authors */
 /** @author Ian Hofmann-Hicks (evil) */
 
-const List = require('../core/List')
 const Pair = require('../core/Pair')
 const isObject = require('../core/isObject')
+const isArray = require('../core/isArray')
 
 // toPairs : Object -> List (Pair String a)
 function toPairs(obj) {
-  if(!isObject(obj)) {
-    throw new TypeError('toPairs: Object required for argument')
+  if(!isObject(obj) && !isArray(obj)) {
+    throw new TypeError('toPairs: Object or Array required for argument')
   }
 
   return Object.keys(obj).reduce(
```

Comment:
can we re-write this using `Object.entries`

```javascript
(acc, [ key, value ]) => ...
```

Reply:
done.

I did the changes without access to a coding environment, so apologies for the high number of commits.